### PR TITLE
Add endpoints to format page view count and page revision edits 

### PR DESCRIPTION
**Pull request by sukritsangvong**
- Add two functions to format the response to timeseries format. 
- integrate functions with graphs so that we can now show the graph of an article that we are looking at. 

Note: Page view data only dates back to May 2015 by the API. I am not sure yet whether this is a Wikipedia thing or an API thing. What we can do is switch API or fixed the max date of looking back in time to May 2015. We can deal with this in a separate PR.

**I have not connected this with the slider yet**

image

Example of API calls:
https://github.com/sukritsangvong/wp-browser-extension/tree/sangvongp/format-timeseries-endpoints#how-to-get-page-count

### CODE REVIEW

**Comment by sukritsangvong** [comment]
note to self: make the x-axis array contain all the dates (including the dates where they is no edit/views) if they are within range

File: README.md [hunk 1]
```diff
@@ -35,44 +35,36 @@ for (var i = 0; i < arrayLength; i++) {
 Use enums.js as inputs for some of the parameters if needed.
 
 ```javascript
```
Comment (sukritsangvong) [inline README.md:37]: Replace old endpoints with new ones

File: wikichange/src/content/content.js [hunk 2]
```diff
@@ -1,78 +1,97 @@
...
-    floatContainer.setAttribute('id', 'floatContainer');
-
-    let graphContainer = document.createElement('div');
-    graphContainer.setAttribute('id', 'graphOverlay');
-
-    let canvas = document.createElement('canvas');
-    canvas.style.maxHeight = '200px';
-    canvas.id = 'viewsEditsChart';
-    graphContainer.style.cssText = 'width:75%;height:20%;';
+const renderGraphOverlay = async () => {
```
Comment (sukritsangvong) [inline wikichange/src/content/content.js:16]: Some of these changes were because of my auto-formatting*

File: wikichange/src/content/content.js [hunk 3]
```diff
@@ -1,78 +1,97 @@
...
-    let p = document.createElement('p');
+    let p = document.createElement("p");
     graphContainer.appendChild(p);
 
-    let siteSub = document.getElementById('siteSub');
+    let siteSub = document.getElementById("siteSub");
     insertAfter(floatContainer, siteSub);
-}
+    const creationDate = await getPageCreationDate();
+    injectGraphToPage(title, creationDate, new Date(Date.now()));
```
Comment (sukritsangvong) [inline wikichange/src/content/content.js:38]: Only added this line

Comment (sukritsangvong) [inline wikichange/src/content/content.js:38]: Added these lines. can change the start date to something that is not `creationDate` later.

File: wikichange/src/content/graph.js [hunk 4]
```diff
@@ -1,74 +1,79 @@
...
+/**
+ * Injects a graph of a given article's page views and edits data.
+ *
+ * @param {string} title of the article
+ * @param {Date} startDate
+ * @param {Date} endDate
+ */
+const injectGraphToPage = async (title, startDate, endDate) => {
+    const pageViews = await getPageViewTimeseries(title, startDate, endDate);
+    const revisions = await getPageRevisionCountTimeseries(title, startDate, endDate);
```
Comment (sukritsangvong) [inline wikichange/src/content/graph.js:23]: Added api calls here

**Review by RichartE — COMMENTED** [review]
The graph looks way too dense, it is hard to see trends, maybe we should group the data a bit or we could try using a thinner line.

**Comment by RichartE** [comment]
For the views pre 2015 we could use this to avoid the 0 in the pink line. https://stackoverflow.com/questions/68692588/chart-js-line-graph-start-a-line-in-the-middle-of-a-graph.

**Review by LuisaE — APPROVED** [review]
Great, thanks!

**Review by RichartE — APPROVED** [review]

**Pull request merged**